### PR TITLE
move fixed json msg to debug_mode

### PR DESCRIPTION
**Pull request by rhernaus**
### Background
Almost every chat interaction displays messages about fixing json that was returned from chat with ai. This PR does not remove these messages, but they will only be showed when debug_mode is enabled.

### Changes
Moved the messages 
`Attempting to fix JSON by finding outermost brackets`
and
`Apparently json was fixed.`
to only show in debug_mode.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by karunakarhv — CHANGES_REQUESTED** [review]
This would work 
```python
if cfg.debug_mode and cfg.speak_mode:
```
instead of 
```python
if cfg.speak_mode and cfg.debug_mode
```

**Review by richbeales — DISMISSED** [review]

File: scripts/main.py [hunk 1]
```diff
@@ -33,9 +33,10 @@ def check_openai_api_key():
 
 
 def attempt_to_fix_json_by_finding_outermost_brackets(json_string):
-    if cfg.speak_mode and cfg.debug_mode:
-      speak.say_text("I have received an invalid JSON response from the OpenAI API. Trying to fix it now.")
-    logger.typewriter_log("Attempting to fix JSON by finding outermost brackets\n")
+    if cfg.debug_mode:
+        if cfg.speak_mode:
+            speak.say_text("I have received an invalid JSON response from the OpenAI API. Trying to fix it now.")
```
Comment (p-i-) [inline scripts/main.py]: This is going in the wrong direction.
`if a and b:` is pythonic, nested isn't.

**Comment by p-i-** [comment]
Is this really getting to the root of the problem tho'?

Sometimes it's valid json, sometimes it needs correcting.

I feel sometimes it's saying 'correcting' when string was in fact valid json.

What we want is a DEBUG log message whenever correction was needed that dumps the string (in a different colour e.g. yellow), so we can see what was wrong.

This will help us prompt-engineer to improve GPT's ability to spit out valid json.

There's other PRs that contain attempts at such prompt-engineering, but without the above logging it's very hit&miss.

Thoughts?

File: scripts/main.py (hunk 1 above)
Comment (rhernaus) [reply scripts/main.py]: More pythonic and following PEP 8 would look like this, making it easier to read and maintain.
```python
    if cfg.debug_mode:
        message = "I have received an invalid JSON response from the OpenAI API. Trying to fix it now."
        log_message = "Attempting to fix JSON by finding outermost brackets\n"

        if cfg.speak_mode:
            speak.say_text(message)
        
        logger.typewriter_log(log_message)
```

**Pull request closed without merging**